File: cpp/leetcode/325.cpp
class Solution {
public:
    int maxSubArrayLen(vector<int> &nums, int k) {
        if (nums.empty()) {
            return 0;
        }
        int sum = 0, res = 0;
        unordered_map<int, int> m{{0, -1}};
        for (int i = 0; i < nums.size(); ++i) {
            sum += nums[i];
            if (sum == k) {
                res = i + 1;
            }
            else if (m.count(sum - k)) {
                res = max(res, i - m[sum - k]);
            }
            if (!m.count(sum)) {
                m[sum] = i;
            }
        }
        return res;
    }
};

File: python/leetcode/325.py
class Solution:
    def maxSubArrayLen(self, nums, k):
        if not nums:
            return 0
        arraySum = {0: -1}
        sums, max_len = 0, 0
        for i in range(len(nums)):
            sums += nums[i]
            if sums not in arraySum:
                arraySum[sums] = i
            if sums - k in arraySum:
                idx = arraySum[sums - k]
                max_len = max(max_len, i - idx)
        return max_len
